staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/helpers.go

```diff
@@ -47,20 +47,11 @@ func LabelSelectorAsSelector(ps *LabelSelector) (labels.Selector, error) {
 		requirements = append(requirements, *r)
 	}
 	for _, expr := range ps.MatchExpressions {
-		var op selection.Operator
-		switch expr.Operator {
-		case LabelSelectorOpIn:
-			op = selection.In
-		case LabelSelectorOpNotIn:
-			op = selection.NotIn
-		case LabelSelectorOpExists:
-			op = selection.Exists
-		case LabelSelectorOpDoesNotExist:
-			op = selection.DoesNotExist
-		default:
+		err, op := selection.TransToOperator(string(expr.Operator))
+		if err != nil {
 			return nil, fmt.Errorf("%q is not a valid pod selector operator", expr.Operator)
 		}
-		r, err := labels.NewRequirement(expr.Key, op, append([]string(nil), expr.Values...))
+		r, err := labels.NewRequirement(expr.Key, *op, append([]string(nil), expr.Values...))
```

Comment:
`NewRequirement` also uses `switch` to justify the type of `operation` after  `transToOperator` already knows the type. I guess if `NewRequirement` also has some room for optimization ？

Reply:
The switch in `NewRequirement ` is likely to verify  values in every Operator, not trans other string to Operator.Did i miss something?